superdesk/io/feed_parsers/nitf.py

```diff
@@ -21,163 +22,245 @@
+                             'filter': self.get_norm_datetime,
+                             },
+            'versioncreated': {'xpath': 'head/docdata/date.issue',
+                               'filter': self.get_norm_datetime,
+                               },
+            'expiry': {'xpath': 'head/docdata/date.expire',
+                       'filter': self.get_norm_datetime,
+                       },
+            'subject': self.get_subjects,
+            'body_html': self.get_content,
+            FORMAT: self.get_format,
+            'place': self.get_place,
+            'keywords': {'xpath': 'head/docdata',
+                         'filter': self.get_keywords,
+                         },
+            'genre': self.get_genre,
+            'ednote': 'head/docdata/ed-msg/@info',
+            'headline': self.get_headline,
+            'abstract': self.get_abstract,
+            'byline': self.get_byline,
+            # metadata
+            'slugline': "head/meta/[@name='anpa-keyword']/@content",
+            'ingest_provider_sequence': "head/meta/[@name='anpa-sequence']/@content",
+            'anpa_category': {'xpath': "head/meta/[@name='anpa-category']",
+                              'filter': lambda elem: [{'qcode': elem.get('content'), 'name': ''}],
+                              },
+            'word_count': {'xpath': "head/meta/[@name='anpa-wordcount']",
+                           'filter': lambda elem: int(elem.get('content')),
+                           },
+            'anpa_take_key': "head/meta/[@name='anpa-keyword']",
+            ITEM_TYPE: {'xpath': "head/meta/[@name='anpa-format']",
+                        'filter': self.anpa_format_filter,
+                        },
+            'priority': {'xpath': "head/meta/[@name='aap-priority']",
+                         'filter': lambda elem: self.map_priority(elem.get('content')),
+                         },
+            'original_creator': self.get_original_creator,
+            'version_creator': self.get_version_creator,
+            'original_source': "head/meta/[@name='aap-source']/@content",
+            'source': "head/meta/[@name='aap-source']/@content",
+            'task': self.get_task,
+        }
+        self.metadata_mapping = None
+
+    def _parse_xpath(self, xpath):
+        """parse xpath and handle final attribute"""
+        last_idx = xpath.rfind('/')
+        last = xpath[last_idx + 1:].rstrip()
+        if last_idx == -1:
+            msg = "No path separator found in xpath {}, ignoring it".format(xpath)
+            logger.warn(msg)
+            raise ValueError(msg)
+        if last.startswith('@'):
+            # an attribute is requested
+            return {'xpath': xpath[:last_idx], 'attribute': last[1:]}
+        else:
+            return {'xpath': xpath}
+
+    def _parse_mapping(self, value):
+        if isinstance(value, dict):
+            if not value:
+                return {}
+            try:
```

Comment:
A shorter version would be:
`if 'xpath' not in value:
    return value`
So you have 2 lines instead of 5